src/app/util/ember-compatibility-functions.cpp

```diff
@@ -386,9 +387,10 @@ CHIP_ERROR ReadSingleClusterData(const SubjectDescriptor & aSubjectDescriptor, b
 
     {
         Access::RequestPath requestPath{ .cluster = aPath.mClusterId, .endpoint = aPath.mEndpointId };
-        Access::Privilege requestPrivilege = Access::Privilege::kView; // TODO: get actual request privilege
-        CHIP_ERROR err                     = Access::GetAccessControl().Check(aSubjectDescriptor, requestPath, requestPrivilege);
-        err                                = CHIP_NO_ERROR; // TODO: remove override
+        Access::Privilege requestPrivilege =
+            Access::RequiredPrivilege::ForReadAttribute(aPath.mClusterId, aPath.mEndpointId, aPath.mAttributeId);
```

Comment:
Similarly, ForReadAttribute/ForWriteAttribute should probably just take ConcreteAttributePath.

Reply:
Same as previous comment: it's a system layering issue.

Reply:
I moved the class from src/access to src/app and made that change.